### ISSUE TITLE
Support parsing typedef in DDS Proxy IDL file.

### ISSUE THREAD

**Issue by OdyWayne** [emqx/nanomq#1475]
[Idl-serial](https://github.com/nanomq/idl-serial) can convert IDL to JSON with less codes.

I've noticed that the typedef parser feature has been in the TODO list for a long time.
I'm looking forward to this feature and hope it can come out in the near future.

**Comment by adarsh-jha-dev** [comment]
Hey @OdyWayne , could you please assign this issue to me?

**Comment by OdyWayne** [comment]
Yes, of course. But since this is the feature we are about to working on, we would expect it to be done as soon as possible.
